geopandas/io/sql.py

```diff
@@ -43,9 +46,20 @@ def read_postgis(sql, con, geom_col='geom', crs=None, index_col=None,
     if geom_col not in df:
         raise ValueError("Query missing geometry column '{}'".format(geom_col))
 
+    def load_geom(x):
+        return shapely.wkb.loads(str(x), hex=hex_encoded)
```

Comment:
I _think_ that getting the converting the postgis dump this way both accomplishes what #546 set out to do, but without the `geoalchemy2` dependency, and also may permit reading geometries from SQLite database, if my reading of #101 is correct (using `hex_encoded=False`). I tried testing out the `geoalchemy2` part a bit, and if I understand the issue correctly, this works. I made no effort at this point to test the SQLite read, since I've never used a SQLite database, but perhaps someone else in the community who does might assist?